### PR TITLE
Enforce compatible pyarrow version, fix import and naming issue

### PR DESCRIPTION
**Pull request by StephanieYuan**
*Description of changes:*
* Enforce installation of compatible pyarrow version 0.15 
* Fix import issue and file naming issue

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "import os\n",
```
Comment (chenliu0831) [inline introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb:21]: <p><code>FILE_NAME</code>could be more accurate? And we can replace Line 20 with the constant as well.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/1144/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (edwardjkim) [inline introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb:21]: <p>Can we change this to <code>python -m pip install</code>  instead? I've encountered problems with the <code>pip</code>  CLI wrapper when I changed notebook kernels. See <a href="https://github.com/pypa/pip/issues/3164#issuecomment-398216682" target="_blank">https://github.com/pypa/pip/issues/3164#issuecomment-398216682</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/1144/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (StephanieYuan) [reply introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb:21]: Good catch, thanks Peter!

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb [hunk 2]
```diff
@@ -63,8 +64,8 @@
     "from sklearn.datasets import load_svmlight_file\n",
     "\n",
     "# Download the dataset and load into a pandas dataframe\n",
-    "FILE_DATA = 'abalone'\n",
-    "urllib.request.urlretrieve(\"https://archive.ics.uci.edu/ml/machine-learning-databases/abalone/abalone.data\", FILE_DATA)\n",
+    "FILE_NAME = 'abalone.csv'\n",
+    "urllib.request.urlretrieve(\"https://archive.ics.uci.edu/ml/machine-learning-databases/abalone/abalone.data\", FILE_NAME)\n",
     "feature_names=['Sex', \n",
```
Comment (edwardjkim) [inline introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb:69]: Thanks for working on fixing this notebook. The notebook as currently written will still fail even with your fixes:
```
ValueError: could not convert string to float: 'F'
```
or
```
ValueError: could not convert string to float: 'M'
```
This is due to the "Sex" column being a categorical column: ["F", "M"]. When I encoded it with
```
data["Sex"] = data["Sex"].astype("category")
data["Sex"] = data["Sex"].cat.codes
```
the training job succeeded and the notebook worked end-to-end. I know this bug was there before you started working on this notebook, but can we fix it while we are at it?

Comment (StephanieYuan) [reply introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb:69]: Addressed, thanks Edward

**Pull request merged**